src/torchmetrics/functional/regression/nrmse.py

```diff
@@ -0,0 +1,100 @@
+# Copyright The Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Tuple, Union
+
+import torch
+from torch import Tensor
+from typing_extensions import Literal
+
+from torchmetrics.functional.regression.mse import _mean_squared_error_update
+
+
+def _normalized_root_mean_squared_error_update(
+    preds: Tensor, target: Tensor, num_outputs: int, normalization: Literal["mean", "range", "std"] = "mean"
+) -> Tuple[Tensor, int, Tensor]:
+    """Updates and returns the sum of squared errors and the number of observations for NRMSE computation.
+
+    Args:
+        preds: Predicted tensor
+        target: Ground truth tensor
+        num_outputs: Number of outputs in multioutput setting
+        normalization: type of normalization to be applied. Choose from "mean", "range", "std"
+
+    """
+    sum_squared_error, num_obs = _mean_squared_error_update(preds, target, num_outputs)
+
+    target = target.view(-1) if num_outputs == 1 else target
+    if normalization == "mean":
```

Comment:
Please could we also have an L2 option which normalises by the sum of squares of target? This is a commonly used NMSE in image processing eg for MRI (eg see Eq. 5 in https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3947978/#sec-a.w.btitle )

Reply:
Hi @Borda , do you need any more input from me regarding this? Thanks

Reply:
@SkafteNicki ^^

Reply:
Support added for l2 in commit https://github.com/Lightning-AI/torchmetrics/pull/2442/commits/069354531ac911ab9e87e58e890d95552db58f27